src/wp-includes/rest-api/class-wp-rest-request.php

```diff
@@ -420,14 +420,29 @@ public function has_param( $key ) {
 	/**
 	 * Sets a parameter on the request.
 	 *
+	 * If the given parameter key exists in any parameter type an update will take place,
+	 * otherwise a new param will be created in the first parameter type (respecting
+	 * get_parameter_order()).
+	 *
 	 * @since 4.4.0
 	 *
 	 * @param string $key   Parameter name.
 	 * @param mixed  $value Parameter value.
 	 */
 	public function set_param( $key, $value ) {
-		$order                             = $this->get_parameter_order();
-		$this->params[ $order[0] ][ $key ] = $value;
+		$order     = $this->get_parameter_order();
+		$found_key = false;
+
+		foreach ( $order as $type ) {
+			if ( 'defaults' !== $type && isset( $this->params[ $type ][ $key ] ) ) {
```

Comment:
Let's switch this to `array_key_exists`. We can have `null` parameter values.

Reply:
good point